### PR TITLE
New Spell: Soul Tap!

### PR DESCRIPTION
**Pull request by tralezab**
I always enjoy the dangerous wizard playstyle

:cl:
add: Wizard has a new spell: Soul tap!
tweak: lichdom requires a soul now, and it takes your soul.
/:cl:

this seems to work fine, i'm making the health you lose depend on how long the cooldowns you refreshed were.

### CODE REVIEW

File: code/modules/spells/spell_types/lichdom.dm [hunk 1]
```diff
@@ -154,3 +155,25 @@
...
 	return "Respawn of [mind] successful."
+
+/obj/effect/proc_holder/spell/self/tap
+	name = "Soul Tap"
+	desc = "Fuel your spells using your own soul! Nope, nothing can go wrong with this."
+	school = "necromancy" //i could see why this wouldn't be necromancy but messing with souls or whatever
+	charge_max = 10
+	invocation = "AT ANY COST"
+	invocation_type = "shout"
+	level_max = 0 //cannot be improved either, mostly because there isn't much to improve beyond how much health you lose and that is already generous for what it does
```
Comment (PKPenguin321) [inline code/modules/spells/spell_types/lichdom.dm]: Improving spells usually just reduces the cooldown

**Comment by lordpidey** [comment]
Consider also checking that they own their soul.

**Comment by tralezab** [comment]
that's the other check i have to add, yeah. it's in the changelog.

**Comment by tralezab** [comment]
is there any problematic spells i need to think about right now? I tested with emp (generic aoe target spell), fireball (aimed held cast), and mutate (active effect). all of them worked fine.

**Comment by Bombpants** [comment]
Wait, what does it do?

**Comment by tralezab** [comment]
removes a certain amount of max life to remove your cooldowns. the higher the combined cooldown, the more max health you lose.

**Comment by Bombpants** [comment]
Mind swapping sounds !FUN! Let the wizards suck everyone’s soul.

**Comment by tralezab** [comment]
siphoning their soul to recharge their mindswap alone would be a no downside 0 cooldown mindswap but also it would be turning the station into a bunch of low-health walking husks

which is HILARIOUS but i don't think it would be fair so i have to work out some way to prevent that

**Comment by Bombpants** [comment]
Maybe offer some cute through the lawyer’s ability to own souls (the thing through the devil mode) to crew who do not know the spell?

**Comment by Iamgoofball** [comment]
Make Devils actually buy your soul

**Comment by tralezab** [comment]
would be a good start to fixing devil

but i don't know if i'd want to put effort into devil just because it's so incredibly hard to fix and so it's most likely just going to be a waste of my time

**Comment by evsey9** [comment]
Does it reduce the cooldowns forever or is it like a one time thing where you can reduce your max health to recharge spell cooldowns a few times and then you just die from two spear hits?

**Comment by tralezab** [comment]
Okay, this is ready. I had a revelation that I could just make soul tap require wizard garbs. That way, wizard memes can happen but they're fair. I have also decided to increase the soul tap cost but not to have it scale off of your spell cooldowns, to support "big wizard" kind of build.

File: code/modules/spells/spell_types/lichdom.dm [hunk 2]
```diff
@@ -154,3 +158,25 @@
 
 
 	return "Respawn of [mind] successful."
+
+/obj/effect/proc_holder/spell/self/tap
```
Comment (kevinz000) [inline code/modules/spells/spell_types/lichdom.dm]: Duplicate definitions?

File: code/modules/spells/spell_types/soultap.dm [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
+//SOUL TAP!//
+//Trades 15 max health for a refresh on all of your spells. I was considering making it depend on the cooldowns of your spells, but I want to support "Big spell wizard" with this loadout.
+//the two spells that sound most problematic with this is mindswap and lichdom, but soul tap requires clothes for mindswap and lichdom takes your soul, so you wouldn't be able to tap at all.
+
+/obj/effect/proc_holder/spell/self/tap
```
Comment (kevinz000) [inline code/modules/spells/spell_types/soultap.dm]: Duplicate definitions?

File: code/modules/spells/spell_types/lichdom.dm [hunk 4]
```diff
@@ -154,3 +158,25 @@
...
+	invocation_type = "shout"
+	level_max = 0 //cannot be improved either, mostly because there isn't much to improve beyond how much health you lose and that is already generous for what it does
+	cooldown_min = 10
+
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "skeleton"
+
+/obj/effect/proc_holder/spell/self/tap/cast(mob/living/user = usr)
+	if(!user.mind.hasSoul) //lol
+		to_chat(user, "<span class='warning'>You do not possess a soul to tap into!</span>")
```
Comment (ChangelingRain) [inline code/modules/spells/spell_types/lichdom.dm]: you need to return after this line

File: code/modules/spells/spell_types/lichdom.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/spells/spell_types/lichdom.dm]: dunno how this got left in, let me handle that

File: code/modules/spells/spell_types/lichdom.dm (hunk 4 above)
Comment (tralezab) [reply code/modules/spells/spell_types/lichdom.dm]: i left in outdated code arrrrgh

File: code/modules/spells/spell_types/soultap.dm [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "soultap"
+
+/obj/effect/proc_holder/spell/self/tap/cast(mob/living/user = usr)
+	if(!user.mind.hasSoul)
+		to_chat(user, "<span class='warning'>You do not possess a soul to tap into!</span>")
+		return
+	to_chat(user, "<span class='danger'>Your body feels drained and there is a burning pain in your chest.</span>")
+	user.maxHealth -= 20
+	user.health = min(user.health - 20, user.maxHealth)
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/soultap.dm]:
```suggestion
	user.health = min(user.health, user.maxHealth)
```
Unless this is intended to always make you lose 20 health?

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
make a define (or variable) for the health lost per cast.

Note:Variable allows admins to customize.

**Comment by optimumtact** [comment]
I am going to close this in 24 hours unless someone else merges or reviews it first.

@tgstation/commit-access

File: code/modules/spells/spell_types/soultap.dm [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/spells/spell_types/soultap.dm]:
```suggestion
#define HEALTH_LOST_PER_SOUL_TAP 20
```

File: code/modules/spells/spell_types/soultap.dm [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "soultap"
+
+/obj/effect/proc_holder/spell/self/tap/cast(mob/living/user = usr)
+	if(!user.mind.hasSoul)
+		to_chat(user, "<span class='warning'>You do not possess a soul to tap into!</span>")
+		return
+	to_chat(user, "<span class='danger'>Your body feels drained and there is a burning pain in your chest.</span>")
+	user.maxHealth -= 20
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/spells/spell_types/soultap.dm]:
```suggestion
	user.maxHealth -= HEALTH_LOST_PER_SOUL_TAP
```

File: code/modules/spells/spell_types/soultap.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [inline code/modules/spells/spell_types/soultap.dm]:
```suggestion
	user.health = min(user.health - HEALTH_LOST_PER_SOUL_TAP, user.maxHealth)
```

File: code/modules/spells/spell_types/soultap.dm (hunk 6 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/spells/spell_types/soultap.dm]: Only if you're going define route instead of variable

Comment (tralezab) [reply code/modules/spells/spell_types/soultap.dm]: great idea

**Comment by k-k-karma** [comment]
nobody will use this and nobody will use 10 max health for fucking spell recharge.

**Comment by tralezab** [comment]
@k-k-karma there are a surprisingly large amount of high cost spells, and it recharges every single one on use. I'm sure people will find a fun build for it.

also chill wow, it's just a pr

**Comment by ExcessiveUseOfCobblestone** [comment]
@k-k-karma dude use it as needed > bodyswap when low > kill the obv very easy to crit old body > take & wear garb > repeat

Reminder: only need hat and robe, no shoes. You can also instant summon one of those.

I think it will be quite viable for those who know how to properly utilize it.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think something went wrong with ur pr lol

**Comment by tralezab** [comment]
Git is just sperging out on the files changed because i had the audacity to fix a merge conflict

**Comment by ExcessiveUseOfCobblestone** [comment]
line ending fun...

**Pull request merged**